src/contexts/product-context.js

```diff
@@ -0,0 +1,161 @@
+import { createContext, useContext, useReducer } from "react";
+import { useAxios } from "../utility/hooks/useAxios";
+import {products} from "../backend/db/products"
+
+const ProductContext = createContext();
+
+const defaultFilter =  {
+  rating: "",
+  material : 
+  {
+    cotton: false,
+    rayon: false,
+    synthetic: false,
+    wool: false
+  },
+  sortBy : "",
+  price: Number.MAX_SAFE_INTEGER
+}
+
+
+const getRatingBadge = rating => {
```

Comment:
Suggestion to move these functions in a separate file like utils and call it here